packages/astro/src/core/config/schema.ts

```diff
@@ -347,10 +347,11 @@ export const AstroConfigSchema = z.object({
 				fallback: z.record(z.string(), z.string()).optional(),
 				routing: z
 					.object({
-						prefixDefaultLocale: z.boolean().default(false),
-						redirectToDefaultLocale: z.boolean().default(true),
-						strategy: z.enum(['pathname']).default('pathname'),
+						prefixDefaultLocale: z.boolean().optional().default(false),
+						redirectToDefaultLocale: z.boolean().optional().default(true),
+						strategy: z.enum(['pathname']).optional().default('pathname'),
 					})
+					.optional()
```

Comment:
The use of `optional()` here is superfluous as `default()` automatically sets the field as optional:

```suggestion
						prefixDefaultLocale: z.boolean().default(false),
						redirectToDefaultLocale: z.boolean().default(true),
						strategy: z.enum(['pathname']).default('pathname'),
					})
```

Here’s a simplified example:

```ts
import { z } from "zod";

const schema = z.object({
	fieldWithDefault: z.string().default("Default"),
	// Identical to the above — `optional()` is superfluous.
	fieldWithOptionalThenDefault: z.string().optional().default("Default"),
});

type input = z.input<typeof schema>;
// type input = {
//     fieldWithDefault?: string | undefined;
//     fieldWithOptionalThenDefault?: string | undefined;
// }

type output = z.output<typeof schema>;
// type output = {
//     fieldWithDefault: string;
//     fieldWithOptionalThenDefault: string;
// }
```
[TS Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcBEyEAJvgNwBQFA9NXAJIB2cMAFgKZYQA23EA7sEYBzOAGcAxhxABDFhzgBXMeyhwAVELCKY6uMDFwZ3fjICehiGBjAIjYwBoadAEY75nYsEyZV7RvCS0jKGrDIAbp7evlD+8BA62vAwZmDsYgB0VBJ2YoFS7LJwALwoGRAuAFbsEjAAFAgUAJDU6urNGnAA8ok6AFxGJuaGcnlQQqJsMowA1oYwEB5wxOyYMorcMFlNGtTNmMDs3MQA6sBsACKr65sDyBljE3UAlBkraxv1+Fcfm-jPThabQ6enoKwCwAkxhYizYnBkLggkTggBQCOAAAysNjsxhe6P0hjEijSUEw3EUCUyIL2TQOR1O51YXWstns3AAKhxGD8bjA7g8YOMRC9yiycdwRe9eXVvtdPv9Aa12js9D0YEkBpAYHFgMZuGYlIx3kJ2MRtrt9odjmc2MzsWz+Y9ha8sazcQDmkqQXBOQYCXASWEwIYQBA8nAcoxMMoJnA7OxzarenyAxBtRC9QbFEbViazd7fYY-S4ausVBjXeK8f6pDI83AAIIAMXZAFEAEpLKWfODhYyKBPegBCIUhmbgjHYkTU-DC8DMCTgpgCiZpdOtjJ5n05-jtbu4jsFT1e3c2Mq3fxdYrZLycaGelAoKTS+kYSRKZS0OgAPM-2BBMHEApZAAPkoWgWFSTgv3gUpGgguBEKwK0GUuOVNgAfgGJ1RAAH0NY1J2IcC6CQ5D6RtJlr2MHduXQmAsPEI8RDgfDs0I00SLI8iN1tajuEYnDWII3MiK4sj11Q1gLxgWi93FQTmLwkSDjE5x0CoP842TD97gSdUfz-ACgOCMD1K0-T3zg9SJJQyiZOwpTxKQyTKPktlaIcpihWEZzENcxl3OMRSfOE9jRM4myXLszd6Lk-iQtjNic1UyKILQCggA)